fix(frontend/backend): prevent Sieve connection error when disabled in settings

Shadow243 opened the pull request.

Shadow243 commented on modules/imap/hm-ews.php:
```diff
@@ -151,7 +151,7 @@ public function get_folder_status($folder) {
                 $result = $this->api->getFolderByDisplayName($folder, Enumeration\DistinguishedFolderIdNameType::MESSAGE_ROOT);
                 if (! $result) {
-                    throw new Exception('Folder not found.');
+                    throw new Exception('Folder '. $folder .' not found.');
```
I'm going to leave a message for @kroky  to confirm my theory about this.

kroky replied:
Looks good

marclaporte commented on language/az.php:
```diff
@@ -411,7 +411,7 @@
     'Added repository' => 'Anbar əlavə edilmişdir',
     'Added server!' => 'Server əlavə edilmişdir',
-    'Are you sure?' => 'Siz əminsiniz?',
+    'Are you sure you wanto to delete this server? ' => 'Siz əminsiniz?',
```
typo

kroky commented on modules/sievefilters/modules.php:
```diff
@@ -1373,31 +1374,32 @@ public function process() {
-                $blocked_list = prepare_sieve_script ($current_script);
-                foreach ($blocked_list as $blocked_sender) {
-                    $blocked_senders[] = $blocked_sender;
+        if($this->module_is_supported('sievefilters')  && $this->user_config->get('enable_sieve_filter_setting', DEFAULT_ENABLE_SIEVE_FILTER)){
```
How would the sievefilters module be not supported when you are in that module? If other modules call this one in their setup files, we should probably move those setup lines to the sievefilters/setup file which is only loaded when the module is loaded.

Shadow243 replied:
@kroky We added the ability to disable a module via the interface(Related PR: https://github.com/cypht-org/cypht/pull/675), even if the module is defined in the `.env` file. Since this update, it's now possible to disable Sieve even if it remains declared in `.env`.

<img width="1122" alt="Screenshot 2025-05-01 at 19 55 42" src="https://github.com/user-attachments/assets/aec2ee7d-a4c2-4f01-845b-80481c158c4c" />

I checked: all calls are coming from this module.

kroky replied:
What does it mean that sieve filter is not enabled via UI? Isn't it the same as the .env - user selects it and the module is not loaded at all or does it mean something else?

Shadow243 replied:
The idea was to provide an extra level of control: even if Sieve is enabled in the .env, we should still be able to disable it through the UI if needed. This way, a user can deactivate Sieve at the settings level without touching the .env file. It acts as a second layer of deactivation. so even if it’s technically active via .env, it becomes inaccessible if turned off in the UI.

kroky replied:
Yes, I understand the UI part. I don't understand the technical implementation. If sieve is not enabled via .env or via settings, then the relevant sieve modules should not be loaded. We shouldn't have many places where we check 'if enabled do this, otherwise do that". Module system is flexible enough to provide checks which modules to load and if modules are loaded, they are executed.

Shadow243 commented:
@kroky I've applied the changes requested at [https://github.com/cypht-org/cypht/commit/18aea3e82694b9a81fc4e02008f0885d1ff77329#r157342289](https://github.com/cypht-org/cypht/commit/18aea3e82694b9a81fc4e02008f0885d1ff77329#r157342289)

kroky commented on modules/api_login/modules.php:
```diff
@@ -1,5 +1,6 @@
 <?php
 
+require_once APP_PATH.'modules/core/handler_modules.php';
```
Please get rid of these includes. Module loading system shouldn't be part of the individual modules' code. Furthermore, core module should always be loaded.

kroky commented on modules/imap_folders/site.js:
```diff
@@ -1,51 +1,79 @@
+    const original_icon = link.querySelector('i');
+    const original_icon_clone = original_icon.cloneNode(true);
+    // 3. Create the spinner
+    const spinner = document.createElement('div');
```
Why do you have all these unrelated updates when the PR is for sieve connection? It is better to have PRs that are self-contained - easier to review, easier to trace in the history.

kroky commented on modules/keyboard_shortcuts/modules.php:
```diff
@@ -61,6 +61,8 @@ public function process() {
             $this->session->record_unsaved('Shortcut updated');
             Hm_Msgs::add('Shortcut updated');
+            $this->save_hm_msgs();
+            Hm_Dispatch::page_redirect('?page=shortcuts');
```
This also seems totally unrelated. Please commit bug fixes to separate PRs.

kroky commented on modules/sievefilters/modules.php:
```diff
@@ -1373,39 +1373,64 @@ public function process() {
 
+class Hm_Handler_check_sieve_configuration extends Hm_Handler_Module {
+    public function process() {
+        if($this->module_is_supported('sievefilters')  && $this->user_config->get('enable_sieve_filter_setting', DEFAULT_ENABLE_SIEVE_FILTER)){
```
You can safely skip 'module_is_supported' here. You are inside sievefilters modules, if this is loaded, then module is supported, yes. Keep the check for config.

Regarding the check from config - I believe it is better to move this elsewhere. If the setting is disabled by the user config, then many of the modules defined here in this file need to be disabled (i.e. not executed). Unfortunately, we cannot do this for all the modules, as the setting itself is a module and triggering that needs to work even if it is unchecked (disabled by the UI). Thus, I thought of doing this in setup.php but we don't have the user config loaded there as there are modules to load the config. Hmm, then we might have no other choice but have all these if user enabled sieve, then execute the module. Maybe you can at least move this to a function and call it immediately at the start of each module process method which checks if the user has enabled it. If not, it returns immediately. This way, we don't need to wrap the whole module code in an if, thus indenting one way further.

Shadow243 commented:
Hi Victor,

I’ve split the changes as you suggested — the updated PR now focuses solely on the Sieve connection fix, keeping it clean and self-contained.

Original PR: #1490
Updated PR: 

1. https://github.com/cypht-org/cypht/pull/1536
2. https://github.com/cypht-org/cypht/pull/1537
3. https://github.com/cypht-org/cypht/pull/1538
4. https://github.com/cypht-org/cypht/pull/1539

Let me know if you have any further suggestions!

Shadow243 commented on modules/sievefilters/modules.php:
```diff
@@ -1373,39 +1408,69 @@ public function process() {
-        Hm_IMAP_List::init($this->user_config, $this->session);
-        $imap_account = Hm_IMAP_List::get($form['imap_server_id'], true);
+        // Sieve may be enabled via .env but disabled in the settings, so we need to verify
+        if($this->user_config->get('enable_sieve_filter_setting', DEFAULT_ENABLE_SIEVE_FILTER)) {
```
Updated
